app/src/lib/git/branch.ts

```diff
@@ -51,3 +58,13 @@ export async function deleteBranch(repository: Repository, branch: Branch, accou
 
   return true
 }
+
+async function checkIfBranchExistsOnRemote(repository: Repository, branch: Branch, account: Account | null, remote: string): Promise<boolean> {
+  const args = [
+    ...gitNetworkArguments,
+    'ls-remote', '--heads', remote, branch.nameWithoutRemote,
+  ]
+  const opts = { env: envForAuthentication(account) }
```

Comment:
I'm not sure if this code duplication is okay.

We can replace passing `account` with passing `opts` instead, but it will come at cost of loosing function signature similar to other ones in this module.